client/lib/domains/utils.js

```diff
@@ -18,3 +24,20 @@ export function getDomainType( domainFromApi ) {
 
 	return domainTypes.MAPPED;
 }
+
+function getDomainNameFromReceiptOrCart( receipt, cart ) {
+	if ( receipt && isEmpty( receipt.failed_purchases ) ) {
+		return map( values( receipt.purchases )[ 0 ], 'meta' )[ 0 ];
```

Comment:
I think we should probably search for the product with `isDomainRegistration`:

```js
import { isDomainRegistration } from 'lib/products-values';

...

const domainRegistration = find( values( receipt.purchases ), isDomainRegistration );
return domainRegistration && domainRegistration.meta;
```

The benefit of this is that it works if the purchases are returned in any order, and doesn't throw a type error if the purchases do not contain a domain.